packages/plugins/src/hardhat/tasks.ts

```diff
@@ -1352,3 +1355,171 @@ export const listProjectsTask = async ({}, hre: HardhatRuntimeEnvironment) => {
+
+  const manager = await getChugSplashManager(
+    signer,
+    parsedConfig.options.projectName
+  )
+
+  const signerAddress = await signer.getAddress()
+  await (
+    await manager.transferProxyOwnership(referenceName, signerAddress)
+  ).wait()
+
+  spinner.succeed(`Proxy ownership claimed by address ${signerAddress}`)
+}
+
+task(TASK_CHUGSPLASH_CLAIM_OWNERSHIP)
+  .addPositionalParam(
+    'configPath',
+    'Path to the ChugSplash config file for the project that owns the target contract'
+  )
+  .addPositionalParam(
+    'referenceName',
+    'Reference name of the contract that should be transferred to you'
+  )
+  .addFlag('silent', "Hide all of ChugSplash's output")
+  .setAction(claimOwnershipTask)
+
+const transferOwnershipTask = async (
+  args: {
+    configPath: string
+    proxyAddress: string
+    silent: boolean
+  },
+  hre
+) => {
+  const { configPath, proxyAddress, silent } = args
+  const provider = hre.ethers.provider
+  const signer = provider.getSigner()
+
+  const spinner = ora({ isSilent: silent })
+  spinner.start('Checking project registration...')
+
+  const parsedConfig = loadParsedChugSplashConfig(configPath)
+
+  // Throw an error if the project has not been registered
+  if (
+    (await isProjectRegistered(signer, parsedConfig.options.projectName)) ===
+    false
+  ) {
+    errorProjectNotRegistered(
+      await getChainId(hre.ethers.provider),
+      hre.network.name,
+      configPath
+    )
+  }
+
+  spinner.succeed('Process registration detected')
+  spinner.start('Checking proxy compatibility...')
+
+  const incompatibleProxyError = `ChugSplash does not support your proxy type.
+  Currently ChugSplash only supports proxies that implement EIP-1967 which yours does not appear to do.
+  If you believe this is a mistake, please reach out to the developers or open an issue on GitHub.`
+
+  // Fetch proxy bytecode and check if it contains the expected EIP-1967 function definitions
+  const iface = new ethers.utils.Interface([
+    'function implementation() external view returns (address)',
+    'function admin() returns (address)',
+    'function upgradeTo(address)',
+    'function changeAdmin(address newAdmin)',
+  ])
+  const bytecode = await provider.getCode(proxyAddress)
+  const checkFunctions = ['implementation', 'admin', 'upgradeTo', 'changeAdmin']
+  for (const func of checkFunctions) {
+    const sigHash = remove0x(iface.getSighash(func))
+    if (!bytecode.includes(sigHash)) {
+      throw new Error(incompatibleProxyError)
+    }
+  }
+
+  // Fetch proxy owner address from storage slot defined by EIP-1967
+  const ownerAddress = BigNumber.from(
+    await provider.getStorageAt(
+      proxyAddress,
+      '0xb53127684a568b3173ae13b9f8a6016e243e63b6e8ee1178d6a717850b5d6103'
+    )
+  ).toHexString()
+
+  // Fetch ChugSplashManager address for this project
+  const managerAddress = await getChugSplashManagerProxyAddress(
+    parsedConfig.options.projectName
+  )
```

Comment:
The `getChugSplashManagerProxyAddress` is synchronous, so no need for `await` here